DR-1857 Add links to view data in Google Console and make styling match Terra more

nmalfroy opened the pull request:
This PR changes a couple of core styling variables to make buttons, etc. blue instead of green.

Oh, I also made the text for the tooltips a little bit bigger

It also adds a bunch of links to view metadata for datasets and snapshots in the Google console.

For the dataset info view:
![image](https://user-images.githubusercontent.com/5633787/152423660-04998297-23cb-42e0-b6ca-e9e9c267f41b.png)

For the snapshot detail view:
![image](https://user-images.githubusercontent.com/5633787/152423730-47909a7f-058f-4560-a13f-aeecc0a615ea.png)

Also worth noting: when viewing the data in the console, it uses the user you're logged into Terra with

Oh, and I also added the tables list back to the dataset info panel until we get the schema view

You can check out what it looks like here:
https://jade-nm.datarepo-dev.broadinstitute.org

nmalfroy committed:
Make the primary color in the TDR UI blue and not green

nmalfroy committed:
Add the ability to pass includes options to snapshot and dataset retrieve calls

nmalfroy committed:
Add links to view data in the google console

...and re-add tables

nmalfroy committed:
Grow the tooltip text

nmalfroy committed:
Add link to the snapshots detail page

tlangs commented:
I don't think there needs to be a tooltip for each table in the tables list in the InfoPanel. A `?` icon that displays the tooltip after the "Tables" header would suffice, I think

tlangs commented on src/components/DetailViewHeader.jsx:
```diff
@@ -43,6 +44,14 @@ const styles = (theme) => ({
     marginBottom: '10px',
+    borderBottom: 0,
+  },
+  button: {
+    backgroundColor: theme.palette.common.link,
+    color: theme.palette.common.white,
+    '&:hover': {
+      backgroundColor: theme.palette.common.link,
+    },
```
How come this isn't in `theme.js`?

nmalfroy replied:
derp...it totally should be

tlangs commented on src/components/dataset/query/sidebar/panels/InfoView.jsx:
```diff
@@ -62,9 +95,43 @@ export class InfoView extends React.PureComponent {
+    const tables = dataset.schema.tables.map((table, i) => (
+      <li key={`${i}`} className={classes.listItem}>
+        {linkToBq && (
+          <TerraTooltip
+            title={`Click to navigate to the Google BigQuery console where you can perform more advanced queries against the ${table.name} dataset table`}
+          >
```
I don't think there needs to be a tooltip for each table in the tables list in the InfoPanel. A ? icon that displays the tooltip after the "Tables" header would suffice, I think

tlangs replied:
Also, I'm getting a TerraToolTip _and_ browser-generated tooltip when I hover

nmalfroy replied:
Gah lemme see about that.  I guess I didn't mind the tooltip over individual tables and it ends up taking less space than having a new ? icon on there.  @lstrano-broad thoughts?

tlangs commented on src/components/dataset/query/sidebar/panels/InfoView.jsx:
```diff
@@ -62,9 +95,43 @@ export class InfoView extends React.PureComponent {
 
+    const linkToBq = dataset.accessInformation?.bigQuery !== undefined;
+    const tables = dataset.schema.tables.map((table, i) => (
+      <li key={`${i}`} className={classes.listItem}>
```
Can we make the table list items have borders, or more contrast with the background?

nmalfroy replied:
sure.  Keep in mind these are gonna go away I think once we have the schema view

snf2ye commented on src/components/snapshot/SnapshotPopup.jsx:
```diff
@@ -95,7 +96,19 @@ export class SnapshotPopup extends React.PureComponent {
+      dispatch(
+        getSnapshotById({
+          snapshotId: snapshot.id,
+          include: [
+            SNAPSHOT_INCLUDE_OPTIONS.SOURCES,
+            SNAPSHOT_INCLUDE_OPTIONS.TABLES,
+            SNAPSHOT_INCLUDE_OPTIONS.RELATIONSHIPS,
+            SNAPSHOT_INCLUDE_OPTIONS.ACCESS_INFORMATION,
+            SNAPSHOT_INCLUDE_OPTIONS.PROFILE,
+            SNAPSHOT_INCLUDE_OPTIONS.DATA_PROJECT,
+          ],
```
Do we already need all of these options?

nmalfroy replied:
I wanted to have them all there to make sure that we get all of the potential information we might need

snf2ye reviewed (approved):
Looks like there are some linter issues - particularly due inserting the "user" prop in alphabetical order to the other props. But, Looks good to me - matches the spec.

nmalfroy committed:
Address PR feedback

nmalfroy committed:
Quiet the linter

nmalfroy committed:
Whoops...didn't mean to commit my docker overrides

nmalfroy committed:
Fix styling so buttons don't go off the screen

...and other minor style tweaks like:
- no drop shadows on buttons
- line up buttons so there's less jostle bouncing between tabs

nmalfroy committed:
Mock changes to match the fact we include a query parameter

tlangs commented on src/components/dataset/query/sidebar/panels/InfoView.jsx:
```diff
@@ -87,14 +149,22 @@ export class InfoView extends React.PureComponent {
+            {linkToBq && (
+              <Grid item xs={12}>
+                <TerraTooltip title="Click to navigate to the Google BigQuery console where you can perform more advanced queries against your dataset tables">
+                  <Button color="primary" variant="contained" disableElevation endIcon={<Launch />}>
```
This button doesn't seem to be aligned with the `Tables` header. Might not be important now, since as you've said, this is going away.

nmalfroy commented:
Yeah, I was thinking it was more like a superscript type thing.  And yeah,
it's gonna go away pretty soon

On Fri, Feb 4, 2022 at 3:06 PM Trevyn Langsford ***@***.***>
wrote:

> ***@***.**** approved this pull request.
> ------------------------------
>
> In src/components/dataset/query/sidebar/panels/InfoView.jsx
> <https://github.com/DataBiosphere/jade-data-repo-ui/pull/1068#discussion_r799776951>
> :
>
> >              <Grid item xs={12}>
>                <Typography variant="h5">{dataset.name}</Typography>
>              </Grid>
> +            {linkToBq && (
> +              <Grid item xs={12}>
> +                <TerraTooltip title="Click to navigate to the Google BigQuery console where you can perform more advanced queries against your dataset tables">
> +                  <Button color="primary" variant="contained" disableElevation endIcon={<Launch />}>
>
> This button doesn't seem to be aligned with the Tables header. Might not
> be important now, since as you've said, this is going away.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/DataBiosphere/jade-data-repo-ui/pull/1068#pullrequestreview-873528293>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABK7N66CB5D7ZZUSPU3O62TUZQWSVANCNFSM5NP6YM4Q>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
